pkg/apis/pipeline/v1beta1/pipeline_validation.go

```diff
@@ -529,3 +531,27 @@ func validateResultsFromMatrixedPipelineTasksNotConsumed(tasks []PipelineTask, f
 	}
 	return errs
 }
+
+// Instead of rejecting the deprecated field entirely, warnPipelineFieldsDeprecation
+// emits a WarningLevel FieldError to notify users that the CRD created contains one
+// or more fields that are deprecated.
+func warnPipelineFieldsDeprecation(ctx context.Context, ps *PipelineSpec) (errs *apis.FieldError) {
+	if ps.Resources != nil {
+		errs = errs.Also(&apis.FieldError{
```

Comment:
We'll probably want to define a helper function (e.g. `DeprecatedError`) that returns a message along the lines of "X field is deprecated" at a warning level. let's remove the reference to the api version, it's not necessary.